rpcs3/CMakeLists.txt

```diff
@@ -188,8 +188,38 @@ set(CMAKE_MODULE_PATH "${RPCS3_SRC_DIR}/cmake_modules")
 find_package(OpenGL REQUIRED)
 find_package(OpenAL REQUIRED)
 if(NOT WITHOUT_LLVM)
-    if (EXISTS "${CMAKE_SOURCE_DIR}/llvmlibs")
-        find_package(LLVM 6.0 CONFIG)
+    find_program(TAR_COMMAND "tar")
+    if ("${CMAKE_SYSTEM}" MATCHES "Linux" AND NOT TAR_COMMAND)
+        message(WARNING "Unable to find the tar command on your path, you will be unable to download the llvm libs.")
+    endif()
+    if (NOT BUILD_LLVM AND "${CMAKE_SYSTEM}" MATCHES "Linux" AND TAR_COMMAND)
+        file(DOWNLOAD https://github.com/RPCS3/llvm/releases/download/continuous-linux-release_60/llvmlibs-linux.tar.gz.sha256 "${CMAKE_SOURCE_DIR}/llvmlibs-linux.tar.gz.sha256")
+        file(STRINGS "${CMAKE_SOURCE_DIR}/llvmlibs-linux.tar.gz.sha256" LLVMLIBS_DOWNLOADED_SHA)
+        
+        if (EXISTS "${CMAKE_SOURCE_DIR}/llvmlibs-linux.tar.gz")
+            file(SHA256 "${CMAKE_SOURCE_DIR}/llvmlibs-linux.tar.gz" LLVMLIBS_LOCAL_SHA)
+            if(NOT ${LLVMLIBS_LOCAL_SHA} STREQUAL ${LLVMLIBS_DOWNLOADED_SHA})
+                file(DOWNLOAD https://github.com/RPCS3/llvm/releases/download/continuous-linux-release_60/llvmlibs-linux.tar.gz "${CMAKE_SOURCE_DIR}/llvmlibs-linux.tar.gz" SHOW_PROGRESS)
+            endif()
+        else()
+            file(DOWNLOAD https://github.com/RPCS3/llvm/releases/download/continuous-linux-release_60/llvmlibs-linux.tar.gz "${CMAKE_SOURCE_DIR}/llvmlibs-linux.tar.gz" SHOW_PROGRESS)
+        endif()
+        
+        file(SHA256 "${CMAKE_SOURCE_DIR}/llvmlibs-linux.tar.gz" LLVMLIBS_LOCAL_SHA)
+        if(NOT ${LLVMLIBS_LOCAL_SHA} STREQUAL ${LLVMLIBS_DOWNLOADED_SHA})
+            message(WARNING "downloaded llvmlibs doesn't match latest sha256sum, building LLVM instead.")
+            set (SHA_MATCHES OFF)
+        endif()
+        
+        # Just check that the file actually downloaded before extracting
+        if (SHA_MATCHES AND EXISTS "${CMAKE_SOURCE_DIR}/llvmlibs-linux.tar.gz")
+            file(REMOVE_RECURSE ${CMAKE_SOURCE_DIR}/llvmlibs)
+            file(MAKE_DIRECTORY ${CMAKE_SOURCE_DIR}/llvmlibs)
+            execute_process(COMMAND ${CMAKE_COMMAND} -E tar xf "../llvmlibs-linux.tar.gz"
+                WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}/llvmlibs")
+            set(LLVM_DIR "${CMAKE_SOURCE_DIR}/llvmlibs/lib/cmake/llvm/")
+            find_package(LLVM 6.0 CONFIG)
+        endif()
     endif()
```

Comment:
this line also has spaces, might wanna correct the whole file to use tabs instead